drivers/input/touchscreen_upper.c

```diff
@@ -105,28 +101,25 @@ static const struct file_operations g_touch_fops =
  * Private Functions
  ****************************************************************************/
 
-static void touch_notify(FAR struct touch_upperhalf_s *upper,
+static void touch_notify(FAR struct touch_openpriv_s *openpriv,
                          pollevent_t eventset)
 {
-  FAR struct pollfd *fd;
+  FAR struct pollfd *fd = openpriv->fds;
   int i;
 
-  for (i = 0; i < CONFIG_INPUT_TOUCHSCREEN_NPOLLWAITERS; i++)
+  fd = openpriv->fds;
+  if (fd)
     {
-      fd = upper->fds[i];
-      if (fd)
+      fd->revents |= (fd->events & eventset);
+      if (fd->revents != 0)
         {
-          fd->revents |= (fd->events & eventset);
-          if (fd->revents != 0)
+          /* report event log */
+
+          int semcount;
+          nxsem_get_value(fd->sem, &semcount);
+          if (semcount < 1)
             {
-              /* report event log */
-
-              int semcount;
-              nxsem_get_value(fd->sem, &semcount);
-              if (semcount < 1)
-                {
-                   nxsem_post(fd->sem);
-                }
+              nxsem_post(fd->sem);
             }
```

Comment:
```suggestion
          nxsem_post(fd->sem);
```

Reply:
done

Reply:
@pkarashchenko check sem count before post, could avoid the poll/read thread wake up multiple time but the fifo is empty.

Reply:
@xiaoxiang781216 I just saw that for joystick case we do not check it, so was thinking that is a common way. Maybe we can unify all input devices to make a check before post.

Reply:
We will provide patch for other input upper half driver.